flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -468,6 +423,59 @@ private static Request createRequest(
 
     private <P extends ResponseBody> CompletableFuture<P> submitRequest(
             String targetAddress, int targetPort, Request httpRequest, JavaType responseType) {
+
+        final SSLHandlerFactory sslHandlerFactory = restConfiguration.getSslHandlerFactory();
+        ChannelInitializer<SocketChannel> initializer =
+                new ChannelInitializer<SocketChannel>() {
```

Comment:
Creating a new `ChannelInitializer` per request sound like perf degradation. Have you measured how much?